api_app/analyzers_manager/observable_analyzers/bgp_ranking.py

```diff
@@ -0,0 +1,102 @@
+import json
+import logging
+
+import requests
+
+from api_app.analyzers_manager import classes
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class BGPRanking(classes.ObservableAnalyzer):
+    """
+    wrapper for https://github.com/D4-project/BGP-Ranking
+    """
+
+    observable_name: str
+    period: int  # optional
+
+    def update(self) -> bool:
+        pass
+
+    def run(self):
+        logger.info("Running BGP_Ranking")
+        urls = {
+            "getASN": "https://bgpranking-ng.circl.lu/ipasn_history/?ip=",
+            "getASNRank": "https://bgpranking-ng.circl.lu/json/asn",
+            "getASNHistory": "https://bgpranking-ng.circl.lu/json/asn_history",
+        }
+        finalresposne = {}
+
+        # get ASN from ip
+        try:
+            response = requests.get(urls["getASN"] + self.observable_name)
+            response.raise_for_status()
+            response = response.json()
+            finalresposne["asn"] = response["response"][
+                list(response["response"].keys())[0]
+            ]["asn"]
```

Comment:
simplify this line and manage existence of the keys accordingly here and not at the end

Reply:
```asn = response.get("response", {}).popitem()[1].get("asn", None)
            if not asn:
                raise AnalyzerRunException("ASN not found in response")```
something like this?